hastory-server/src/Data/Hastory/Server/Handler/Entries.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Handler.Entries where
+
+import Data.Hastory.Server.Handler.Import
+
+createEntryHandler :: ServerSettings -> AuthResult AuthCookie -> SyncRequest -> Handler NoContent
+createEntryHandler _ BadPassword _ = unAuthenticated
```

Comment:
There's a nice trick for this. You can move the `AuthResult` out of here using the `ThrowAll` class. See  https://github.com/NorfairKing/smos/blob/10c32134f9eec36af5a17941abb8f417933479d3/smos-server/src/Smos/Server/Serve.hs#L111https://github.com/NorfairKing/smos/blob/10c32134f9eec36af5a17941abb8f417933479d3/smos-server/src/Smos/Server/Serve.hs#L132